ast/annotations_test.go

```diff
@@ -1095,6 +1097,319 @@ func TestAnnotations_toObject(t *testing.T) {
+      "type": "var",
+      "value": "data"
+    },
+    {
+      "location": {
+        "col": 9,
+        "file": "",
+        "row": 14
+      },
+      "type": "string",
+      "value": "test"
+    }
+  ]
+}`,
+				`{
+  "annotations": {
+    "authors": [
+      {
+        "name": "doc"
+      }
+    ],
+    "custom": {
+      "doc": "doc"
+    },
+    "description": "doc",
+    "location": {
+      "col": 1,
+      "file": "",
+      "row": 16
+    },
+    "organizations": [
+      "doc"
+    ],
+    "related_resources": [
+      {
+        "ref": "https://doc"
+      }
+    ],
+    "schemas": [
+      {
+        "definition": {
+          "type": "integer"
+        },
+        "path": [
+          {
+            "type": "var",
+            "value": "input"
+          },
+          {
+            "type": "string",
+            "value": "bar"
+          }
+        ]
+      }
+    ],
+    "scope": "document",
+    "title": "doc"
+  },
+  "location": {
+    "col": 1,
+    "file": "",
+    "row": 44
+  },
+  "path": [
+    {
+      "location": {
+        "col": 9,
+        "file": "",
+        "row": 14
+      },
+      "type": "var",
+      "value": "data"
+    },
+    {
+      "location": {
+        "col": 9,
+        "file": "",
+        "row": 14
+      },
+      "type": "string",
+      "value": "test"
+    },
+    {
+      "type": "string",
+      "value": "p"
+    }
+  ]
+}`,
+				`{
+  "annotations": {
+    "authors": [
+      {
+        "name": "rule"
+      }
+    ],
+    "custom": {
+      "rule": "rule"
+    },
+    "description": "rule",
+    "location": {
+      "col": 1,
+      "file": "",
+      "row": 31
+    },
+    "organizations": [
+      "rule"
+    ],
+    "related_resources": [
+      {
+        "ref": "https://rule"
+      }
+    ],
+    "schemas": [
+      {
+        "definition": {
+          "type": "string"
+        },
+        "path": [
+          {
+            "type": "var",
+            "value": "input"
+          },
+          {
+            "type": "string",
+            "value": "baz"
+          }
+        ]
+      }
+    ],
+    "scope": "rule",
+    "title": "rule"
+  },
+  "location": {
+    "col": 1,
+    "file": "",
+    "row": 44
+  },
+  "path": [
+    {
+      "location": {
+        "col": 9,
+        "file": "",
+        "row": 14
+      },
+      "type": "var",
+      "value": "data"
+    },
+    {
+      "location": {
+        "col": 9,
+        "file": "",
+        "row": 14
+      },
+      "type": "string",
+      "value": "test"
+    },
+    {
+      "type": "string",
+      "value": "p"
+    }
+  ]
+}`,
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.note, func(t *testing.T) {
+			module := MustParseModuleWithOpts(tc.module, tc.options)
+
+			for i, a := range module.Annotations {
+				ref := NewAnnotationsRef(a)
+
+				asJSON := toJSON(ref)
+
+				assert.JSONEq(t, tc.expected[i], asJSON, "serialized AnnotationsRef differs, expected:\n%s\n got\n%s\n", tc.expected[i], asJSON)
```

Comment:
OPA's depending on `github.com/stretchr/testify` is, unfortunately, rather misleading. I will try and explain why we don't use it for such assertions.

The package is in our go.mod since we vendored and altered gojsonschema. This code came with some tests which used testify. This means that the only uses of testify are in this internal package. Ideally these tests wouldn't introduce additional dependencies to OPA.

Sorry that this likely isn't at all clear. I've added it to my todos as I'd like to not mislead future contributors in this way.

In our tests, we generally compare the JSON as strings, e.g. `ast/marshal_test.go`. Do you think you'd be able to do that here too?